src/types.ts

```diff
@@ -1,6 +1,6 @@
+type QuizStatusType = "notStarted" | "inProgress" | "completed";
 type DifficultyType = "easy" | "medium" | "hard";
 type QuestionType = "multiple" | "boolean";
-type QuizStatusType = "notStarted" | "inProgress" | "completed";
 
 interface ICategoryInfo {
```

Comment:
my bad, i gave a wrong suggestion, generic shared types can live here, however component level types should be in the component file

Reply:
Is it strongly recommended that component types be included in the component file or can I leave them in the types file?

I actually prefer having all types and interfaces in one file. It makes the component file shorter and also reduces the need to import extra types. E.g my ISingleQuestionProps interface spans 10 lines, that's 10 extra lines that would be in the SingleQuestion component, but also, I'd need to import all the types that this interface depends on.

Reply:
I understand, however, in the long term, as the project grows, it's important to decouple components and their types into separate files. This will help improve maintainability, scalability, and make the codebase easier to manage in the long run.